src/metabolic_ninja/resources.py

```diff
@@ -185,6 +186,147 @@ def get(self):
         return PRODUCT_LIST
 
 
+class PathwayResource(MethodResource):
+    def get(self, job_id, pathway_id):
+        job_id = int(job_id)
+        try:
+            job = (
+                DesignJob.query.filter(DesignJob.id == job_id)
+                .filter(
+                    DesignJob.project_id.in_(g.jwt_claims["prj"])
+                    | DesignJob.project_id.is_(None)
+                )
+                .one()
+            )
+        except NoResultFound:
+            return (
+                {"error": f"Cannot find any design job with id {job_id}."},
+                404,
+            )
+        else:
+            if job.is_complete():
+                status = 200
+            else:
+                status = 202
+            pathway = next(
+                (
+                    x
+                    for x in job.result["diff_fva"]
+                    + job.result["opt_gene"]
+                    + job.result["cofactor_swap"]
+                    if x["id"] == pathway_id
+                ),
+                None,
+            )
+            if not pathway:
+                return (
+                    {"error": f"Cannot find any job result with id {pathway_id}."},
+                    404,
+                )
+            result = get_tabular_data(pathway)
+            response = make_response(result, status)
+            response.headers["Content-Type"] = "text/csv"
+            response.headers["Content-Disposition"] = "attachment"
+            return response
+
+
+def get_tabular_data(pathway):
+    result = []
+    targets = pathway["targets"]
+    manipulations = pathway["manipulations"]
+    if pathway["method"] == "PathwayPredictor+DifferentialFVA":
```

Comment:
Please create separate functions for each method that are then called by this main function.